documentation/docs/03-routing.md

```diff
@@ -45,206 +57,170 @@ A file called either `src/routes/about.svelte` or `src/routes/about/index.svelte
 
 To get the raw data instead of the page, you can include an `accept: application/json` header in the request, or — for convenience — append `/__data.json` to the URL, e.g. `/items/[id]/__data.json`.
 
-#### Standalone endpoints
+The different types of request handlers and their capabilities are explained in more detail in the following sections.
 
-Most commonly, endpoints exist to provide data to the page with which they're paired. They can, however, exist separately from pages. Standalone endpoints have slightly more flexibility over the returned `body` type — in addition to objects and [`Error`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) instances, they can return a [`Uint8Array`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array) or a [`ReadableStream`](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream).
+#### GET
 
-Standalone endpoints can be given a file extension if desired, or accessed directly if not:
+`GET` handlers return an object representing the response, which must be a plain object that is JSON-serializable. You can optionally set headers using the `setHeaders` method:
 
-| filename                      | endpoint   |
-| ----------------------------- | ---------- |
-| src/routes/data/index.json.js | /data.json |
-| src/routes/data.json.js       | /data.json |
-| src/routes/data/index.js      | /data      |
-| src/routes/data.js            | /data      |
+```js
+/// file: src/routes/random/+page.server.js
+/** @type {import('@sveltejs/kit').RequestHandler} */
+export async function GET({ setHeaders }) {
+	setHeaders({
+		'access-control-allow-origin': '*'
+	});
 
-#### POST, PUT, PATCH, DELETE
+	return {
+		number: Math.random()
+	};
+}
+```
 
-Endpoints can handle any HTTP method — not just `GET` — by exporting the corresponding function:
+If something unexpected happens, you `throw` using the `error` helper:
 
 ```js
-// @noErrors
-export function POST(event) {...}
-export function PUT(event) {...}
-export function PATCH(event) {...}
-export function DELETE(event) {...}
+/// file: src/routes/dan/+page.server.js
+import { error } from '@sveltejs/kit/data';
+
+/** @type {import('./$types').GET} */
+export function GET({ session }) {
+	if (session.user?.name !== 'Dan') {
+		throw error(403, `If your name's not Dan, you're not coming in`);
+	}
+
+	return {
+		answer: 42
+	};
+}
 ```
 
-These functions can, like `GET`, return a `body` that will be passed to the page as props. Whereas 4xx/5xx responses from `GET` will result in an error page rendering, similar responses to non-GET requests do not, allowing you to do things like render form validation errors:
+The first argument to `error` is an [HTTP status code](https://httpstatusdogs.com) (you'll most likely reach for `4xx` — client error, or `5xx` — server error), the second is an explanatory message. You can access both using the [page store](/docs/modules#$app-stores-page).
```

Comment:
Can we please not? I think link to <https://developer.mozilla.org/en-US/docs/Web/HTTP/Status> is better. I agree it's good to have some jokes and funny stuff in docs, but not for the sake of quality. It's needed to people know about what code to use.

Sry I'm funsucker.

Reply:
This is from the old docs, I didn't realize it links to this site 😄 since it's there and noone else has complained I'm inclined to keep it for now